client/src/firebaseInit.js

```diff
@@ -0,0 +1,56 @@
+import firebase from "firebase/compat/app";
+import { getMessaging, getToken, onMessage } from "firebase/messaging";
+import { toast } from "react-hot-toast";
+
+const config = {
+  apiKey: "AIzaSyAgUY5d0kc556sShMaMogBmfcLRNZ_ClDU",
+  authDomain: "studydash-fd048.firebaseapp.com",
+  projectId: "studydash-fd048",
+  storageBucket: "studydash-fd048.appspot.com",
+  messagingSenderId: "66459387705",
+  appId: "1:66459387705:web:8e01b1649d8fd481dd8ae8",
+  measurementId: "G-ELZW5L303E",
+};
+
+const app = firebase.initializeApp(config);
+const messaging = getMessaging(app);
+
+// next block of code goes here
+export const fetchToken = async (setToken) => {
+  await Notification.requestPermission().then((permission) => {
```

Comment:
I recall you mentioned that firebase doesn't work in incognito so this is probably related to that. On Chrome you are unable to provide notification perms in incog but in Edge you can which leads to this error when you do:

![image](https://github.com/awctw/study-dash/assets/56493743/1252b947-0eba-4963-b453-49f0c7200cb4)